### PR TITLE
:full_moon: Integration with Moonsama v0

### PR DESCRIPTION
**Pull request by vikiival**
- :zap: moonsama url should be present
- :tada: copypasted moonsama from bsx
- :zap: expect that default indexer is subsquid
- :wrench: moonsama indexer config
- :top: moonsama detail :)
- :bug: metadata can override id
- :zap: added click
- :tada: copy-paste moonsama here
- :zap: correct build platform for moonsama
- :coffin: removed available actions
- :zap: formatter to evm addr
- :zap: utiiiiils
- :zap: create only visible based on the configuration

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [ ] PR ref #3161

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

### Known problems:

- [ ] Identicon does not work
- [ ] Auth shows Address in polkadot format (Account32) -> should be ETH addr (Account20) //there is util ready for that
- [ ] Collection meta are not visible (my bad, forgot to index them)
- [ ] ERC-1155 are not there yet (kodadot/click#5)
- [ ] Bvlck collection has strange ABI (kodadot/click#6)
- [ ] If you found something else just open issue.
- [ ] Code is copy-paste from basilisk so if you spot some mismatch pls comment

### CODE REVIEW

**Comment by roiLeo** [comment]
Evm (Metamask) wallet integration is more or less ready if you want to test it out

File: pages/moonsama/explore.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<template>
+  <ExploreLayout type="rmrk" />
```
Comment (roiLeo) [inline pages/moonsama/explore.vue:2]: :worried:

File: pages/moonsama/explore.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      url: '/moonsama/explore',
+      image: `${this.$config.baseUrl}/k_card_collections.png`,
+    }
+    return {
+      title,
+      meta: [...this.$seoMeta(metaData)],
+    }
+  },
+})
+export default class ExploreBsx extends Vue {}
```
Comment (roiLeo) [inline pages/moonsama/explore.vue:30]: `ExploreBsx`
:worried:

File: utils/account.ts [hunk 3]
```diff
@@ -1,7 +1,12 @@
 import { KeyringAccount } from '@/utils/types/types'
 import keyring from '@polkadot/ui-keyring'
 import { getAddress } from '@/utils/extension'
-import { decodeAddress, encodeAddress, addressEq } from '@polkadot/util-crypto'
+import {
+  decodeAddress,
+  encodeAddress,
+  addressEq,
+  addressToEvm,
```
Comment (roiLeo) [inline utils/account.ts:8]: didn't knew about this one

File: utils/config/permision.config.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Prefix } from './indexer.config'
```
Comment (roiLeo) [inline utils/config/permision.config.ts:1]: What this file will be used for?

File: utils/queryPathResolver.ts [hunk 5]
```diff
@@ -34,7 +34,7 @@ function getPath(prefix: string) {
     case 'subsquid':
       return 'subsquid/general/'
     default:
-      return ''
+      return 'subsquid/general/'
```
Comment (roiLeo) [inline utils/queryPathResolver.ts:37]: I feel that this line might break something

**Comment by yangwao** [comment]
Seems pagination and infinity scrolling is broken 👀 

but manually adding pages works 

image

File: utils/account.ts (hunk 3 above)
Comment (vikiival) [reply utils/account.ts:8]: People were able to claim KSM/DOT from DOT ICO that happened on ETH

File: utils/config/permision.config.ts (hunk 4 above)
Comment (vikiival) [reply utils/config/permision.config.ts:1]: `(permision or indexer).config ?`

Comment (vikiival) [reply utils/config/permision.config.ts:1]: But `permision.config` -> should mark some features that are not available on some chains -> e.g create on moonsama

`indexer.config` -> {
  one configuration to correctly handle indexers
 }

File: utils/queryPathResolver.ts (hunk 5 above)
Comment (roiLeo) [reply utils/queryPathResolver.ts:37]: This looks like it will break Magick queries

Comment (vikiival) [reply utils/queryPathResolver.ts]: Thanks for info Leo ❤️ fixed in ed41b76040b2c76fa9b2a3a32083282547e6c9a8

**Comment by yangwao** [comment]
https://deploy-preview-3340--koda-nuxt.netlify.app/moonsama/u/0x495E889d1A6cEB447a57dcc1C68410299392380c?tab=sold

I guess hence we have quite long work to do, so let's continue through 

- #3161 

image

**Comment by roiLeo** [comment]
artworks-000528979299-op8qu5-t500x500

**Pull request merged**